### PR TITLE
Build in Circle CI 2.0

### PR DESCRIPTION
**Pull request by fassko**
* increase watchOS minimum version because of RxSwift
* remove Pods from Demo project to build in CI

### CODE REVIEW

**Comment by freak4pc** [comment]
Hey @fassko, thanks ! 
Not sure why but a rebase isn't possible (conflicting) - could you rebase on top of master? 

Also, I'm wondering why are there so many changes if all was needed is a new config? Is it the Pods removal I imagine?

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      - save_cache:
+          key: v2-dep-{{ .Branch }}-{{ epoch }}
+          paths:
+            - Carthage
+workflows:
+  version: 2
+  nightly:
+     triggers:
+       - schedule:
+           cron: "0 0 * * *"
```
Comment (freak4pc) [inline .circleci/config.yml]: I don't think we need a nightly build for this repo...

**Review by freak4pc — CHANGES_REQUESTED** [review]

**Comment by fassko** [comment]
@freak4pc I had to remove Pods because there is some problem with other machines like Circle CI. Also it will make repo smaller.

I will try to rebase.

**Comment by fassko** [comment]
I did rebase from master.

**Comment by freak4pc** [comment]
Uhm doesn't seem like it ... can you take a look again? Worst case I can just merge it but I prefer a rebase: 
image

**Comment by freak4pc** [comment]
Not sure why but this is failing on master. Any thoughts? https://circleci.com/gh/RxSwiftCommunity/NSObject-Rx/62

**Comment by fassko** [comment]
Hmm something is weird with keys. Do you have access to settings of this repo? To see if key which is in Circle is valid?

**Comment by freak4pc** [comment]
OK, passing now. Thanks @fassko ! 

I'm a bit confused by two things:

1. Why do we need both CocoaPods and Carthage for CI? More specifically it seems like Carthage isn't being used at all. Is it there just to confirm it's building?

2. Seems we're only testing iOS... Would be good to do the other supported platforms as well.

**Comment by fassko** [comment]
1. I think you're right Carthage is needed only for local development if you want to edit Xcode project. I will remove it to speed up the build.

2. True, this is very good recommendation.

I will create issues for both of these things otherwise I forget things quickly. :P

**Pull request merged**